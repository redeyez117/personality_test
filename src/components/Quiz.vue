<template>
  <div class="quiz-container">
    <QuizQuestions v-if="!quizResult" @incrementIndex="nextQuestion($event)" :question="questions[currentIndex]"/>
    <div v-else class="result">
      <img :src="resultImg"/>
      <h3>{{ result }}</h3>
      <button @click="$emit('closeQuiz')">Try Again!</button>
    </div>
  </div>
</template>

<script>
import QuizQuestions from "@/components/QuizQuestions";
import {questions} from "@/data/questions";

export default {
  name: "QuizContainer",
  components: {QuizQuestions},
  data() {
    return {
      currentIndex: 0,
      questions,
      quizResult: false,
      totalPoints: 0,
    }
  },
  methods: {
    nextQuestion(points) {
      if (this.currentIndex < this.questions.length) {
        this.totalPoints = this.totalPoints + points
        this.currentIndex++
        if (this.currentIndex === this.questions.length) {
          this.totalPoints = this.totalPoints + points
          this.quizResult = true
        }
      }
    },
  },
  computed: {
    result() {
      if (this.totalPoints <= 80) {
        return 'You are an EXTRAVERT!'
      }
      return 'You are an INTROVERT!'
    },
    resultImg() {
      if (this.totalPoints <= 80) {
        return require('../assets/images/extravert.gif')
      }
      return require('../assets/images/introvert.gif')
    },
  }
}
</script>

<style scoped>

.quiz-container {
  max-width: 900px;
  width: 100%;
  margin-inline: auto;
  padding: 10px;
  background: white;
  border-radius: 5px;
  margin-top: 7rem;
}

.result {
  text-align: center;
}

.result img {
  margin-top: 40px;
  width: 80px
}

.result h3 {
  margin-top: 15px;
  font-size: 2.3rem;
}

.result button {
  padding: 7px 10px;
  border-radius: 6px;
  background-color: lightsteelblue;
  color: white;
  font-size: 18px;
  font-weight: 600;
}
</style>