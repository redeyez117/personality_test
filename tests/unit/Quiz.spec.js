import Quiz from "@/components/Quiz";
import {mount} from "@vue/test-utils";

describe('Quiz.vue', () => {
    it('It renders component ', function () {
        const wrapper = mount(Quiz)
        expect(wrapper.element).toMatchSnapshot();
    });

    it('has initial index of 0', ()=> {
        const wrapper = mount(Quiz)
        expect(wrapper.vm.currentIndex).toEqual(0)
    })


    it('increments index by 1', ()=> {
        const wrapper = mount(Quiz)
        wrapper.vm.nextQuestion()
        expect(wrapper.vm.currentIndex).toEqual(1)
    })
})
