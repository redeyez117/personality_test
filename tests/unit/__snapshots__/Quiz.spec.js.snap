// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quiz.vue It renders component  1`] = `
<div
  class="quiz-container"
>
  
  <h3>
    You’re really busy at work and a colleague is telling you their life story and personal woes. You:
  </h3>
  <div
    class="answer-container"
  >
    <ul
      class="answer-list"
    >
      
      <li>
        Don’t dare to interrupt them
      </li>
      <li>
        Think it’s more important to give them some of your time; work can wait
      </li>
      <li>
        Listen, but with only with half an ear
      </li>
      <li>
        Interrupt and explain that you are really busy at the moment
      </li>
      
    </ul>
  </div>
  
</div>
`;
